Site Appearance (with changes)

Alec-Shay opened the pull request:
Hey @dsanz,

Here is the branch we've been working off of, with my more recent changes. As with the other PR, we can use this space to discuss as needed. Thank you for your help with all of this so far.

Commit:
LRDOCS-7788 Search Upgrade Paths and Backing Up Elasticsearch

Commit:
initial md files

Commit:
wordsmithing

JamesAGarcia committed:
Introduction to the Recycling Bin - Wordsmithing

JamesAGarcia committed:
Correction to the intro Recyling Bin article

JamesAGarcia committed:
LRDOCS-8565 article update with new images

Commit:
few changes

Alec-Shay committed:
LRDOCS-8580 Auto-completion for variables and taglibs in page fragment editor

Alec-Shay committed:
LRDOCS-8581 Document option to make page fragments cacheable

Alec-Shay committed:
LRDOCS-8581 Fix article title and images folder to suit convention (article title should match within the article, file name, and folder)

JamesAGarcia committed:
LRDOCS-8564 new article with images: adding validation rules

JamesAGarcia committed:
LRDOCS-8564 updated rst with new article

JamesAGarcia committed:
LRDOCS-8564 includes admonition about escaping characters in regex values

rbohl committed:
LRDOCS-7788 nav work

rbohl committed:
LRDOCS-7788 rename the Search Upgrade Paths article to "Upgrading Search to 7.3 Overview". I dislike the name, may need to revisit, but it's more specific since this is all about going to 7.3

rbohl committed:
LRDOCS-7788 reworking content for upgrade paths

rbohl committed:
LRDOCS-7788 ported in the liferay-docs article and did some editing

rbohl committed:
LRDOCS-7788 editing

rbohl committed:
LRDOCS-7788 rename ported file to Upgrading to Elasticsearch 7.9

rbohl committed:
LRDOCS-7788 rename "Upgrading Search to 7.3 Overview" to "Upgrading Search for Liferay 7.3"

rbohl committed:
LRDOCS-7788 nav work, deleting unnecessary files

rbohl committed:
LRDOCS-7788 moved the search tuning index backup and restore info to the backing up article

rbohl committed:
LRDOCS-7788 more editing

Alec-Shay committed:
LRDOCS-8565 Minor edits

rbohl committed:
LRDOCS-7788 wordsmithing

rbohl committed:
LRDOCS-7788 rst ref syntax correction

Alec-Shay committed:
LRDOCS-8576 Add tip pointing out button to hide/show widget controls on widget pages

Alec-Shay committed:
LRDOCS-8577 Add tip noting compatibility of Custom Fields with navigation menus

JamesAGarcia committed:
LRDOCS-8564 dataType addition per SME review

Alec-Shay committed:
LRDOCS-8574 Add note about Support for using Global Navigation Menus

Alec-Shay committed:
LRDOCS-8564 Wordsmithing

rbohl committed:
LRDOCS-7782 worked on the navigation a bit and renmed: let's just name these after the apps themselves, Synonym Sets and Result Rankings

rbohl committed:
LRDOCS-7782 and LRDOCS-7783 editing and wordsmithing

lipusz committed:
LRDOCS-7788 SME edits

lipusz committed:
LRDOCS-7788 SME edits

- https://issues.liferay.com/browse/LPS-103938 is no longer reproducible

lipusz committed:
LRDOCS-7788 Fix link

rbohl committed:
LRDOCS-7788 fix links and add workflow metrics re-indexing as a separate step

rbohl committed:
LRDOCS-7788 sme edits and add a step to test the upgraded search experience

Commit:
LRDOCS-8665 portal-developer-properties.md reference

Commit:
LRDOCS-8665 add to rst

rbohl committed:
LRDOCS-7788 less specific version details

rbohl committed:
LRDOCS-8667 this admonition belongs in the article intro

rbohl committed:
LRDOCS-8667 i think we should just combine these notes

lipusz committed:
LRDOCS-8667 Fix securing Elasticsearch steps

Plus some clean-up.

lipusz committed:
LRDOCS-8667 Add more clarity

lipusz committed:
LRDOCS-8667 SF

lipusz committed:
LRDOCS-8667 Add command for PEM format cert generation

lipusz committed:
LRDOCS-8667 Add security settings for 7.3 plus some refactoring

lipusz committed:
LRDOCS-8667 Move deprecation warning section to the troubleshooting article

Signed-off-by: Tibor Lipusz <tibor.lipusz@liferay.com>

lipusz committed:
LRDOCS-8667 Add reference to the securing Elasticsearch article from other articles

Signed-off-by: Tibor Lipusz <tibor.lipusz@liferay.com>

lipusz committed:
LRDOCS-8667 Add friendly url to LES for 7.2 downloads

https://customer.liferay.com/downloads/-/download/liferay-enterprise-search-for-liferay-dxp-7-2

lipusz committed:
LRDOCS-8667 Consistency

Signed-off-by: Tibor Lipusz <tibor.lipusz@liferay.com>

rbohl committed:
LRDOCS-7788 minor editing

jorge-campo committed:
Remove unused README.rst file and update cross-link to the README.rst/index.html.

Commit:
LRDOCS-8686 deprecate JAAS support

Commit:
LRDOCS-8686 Remove references to JAAS support

rbohl committed:
LRDOCS-8667 editing

rbohl committed:
LRDOCS-7788 wordsmithing and testing

Commit:
LRDOCS-8706 Refer to com.acme.greeter. Mention add button. Mention using names from App Manager.

rbohl committed:
LRDOCS-8667 updated cert/key gen

lipusz committed:
LRDOCS-7788 SME edits vol.3.

lipusz committed:
LRDOCS-7788 SME edits vol.3.

lipusz committed:
LRDOCS-7788 Use 7.9+ instead of 7.9.0+

rbohl committed:
LRDOCS-7788 fine tuning

Commit:
LRDOCS-8724 fix wildfly standalone.xml element to jsp-config

rbohl committed:
LRDOCS-7788 cleanup and snapshot note proliferation

rbohl committed:
LRDOCS-7788 more cleanup, argh

Commit:
LRDOCS-8724 remove unneeded jsp-configuration element from JBoss

Commit:
LRDOCS-8725 Add a resource-root element for each JAR from the Liferay dependencies ZIP

Commit:
LRDOCS-8693 move screens from maintenance mode to deprecated

Commit:
LRDOCS-8733 Update tutorial for liferay-r2f1.zip

sez11a committed:
LRDOCS-8565 Wordsmithing.

sez11a committed:
LRDOCS-8564 Wordsmithing.

sez11a committed:
LRDOCS-8571 Wordsmithing.

Commit:
LRDOCS-8733 Update tutorial with liferay-n8g6.zip code changes. Added references to portal source code too.

sez11a committed:
LRDOCS-7782 Wordsmithing.

sez11a committed:
LRDOCS-7782 Wordsmithing.

sez11a committed:
LRDOCS-8665 Wordsmithing.

sez11a committed:
LRDOCS-8647 Wordsmithing.

rbohl committed:
LRDOCS-7788 some changes borught about by the SME's presentation to support:
1. Backing Up: use the API call format that would be used in Kibana instead of cURL
   calls
2. Backing Up: fix the `_restore` api call
3. Backing Up: move the note on using Kiubana to the top of the article
4. Upgrading to 7.3 article: add a callout to Breaking Changes since one
   in particular can impact the search upgrade.

sez11a committed:
LRDOCS-7788 Wordsmithing.

sez11a committed:
LRDOCS-7788 Wordsmithing.

sez11a committed:
LRDOCS-7788 Wordsmithing.

sez11a committed:
LRDOCS-8667 Wordsmithing.

Commit:
Incoming and outgoing ports need to be switched for mail setup.

jrhoun committed:
Fix a variety of links and build errors

Commit:
LRDOCS-8760 Fragment import requires each package has a name

brianchandotcom committed:
LRDOCS-8760 Jim, sort, thx.

jrhoun committed:
LRDOCS-8565 Rename article to be an overview, update links, remove user-guide level, reorganize apis section under developer-guide

jrhoun committed:
LRDOCS-8571 Fix admonition

jrhoun committed:
LRDOCS-7788 Fix links

Commit:
DXP Cloud Portal Properties, extended explanation and example.

Commit:
portal-clu.properties no longer used in v4.0

JamesAGarcia committed:
LRDOCS-8714 UI updates to Staging, replacing 'publication' with 'publish process'

jrhoun committed:
LRDOCS-7782 Wordsmithing, add annotations to screenshots to make it more obvious where to look

jrhoun committed:
LRDOCS-8665 Add link to portal dev properties doc from portal properties, remove readme rst

Alec-Shay committed:
LRDOCS-8768 Reorganize logical flow (and typo fix)

jrhoun committed:
LRDOCS-7782 Wordsmithing

Alec-Shay committed:
Reorganize Site Appearance section and add some theme-related articles

Alec-Shay committed:
Changes and fixes per feedback

dsanz committed:
IFI-2311 Add the "Getting started" section skeleton with references to the corresponding tickets

dsanz committed:
IFI-2311 Move the "creating a theme" article to the getting started section

dsanz committed:
IFI-2311 Complete the "styling" content skeleton to make it match with IFI-2293 requirements

dsanz committed:
IFI-2311 Move the injecting variables article to the (future) templating folder

dsanz committed:
IFI-2311 Rename designing-pages to templating to host all template-related articles from IFI-2294

dsanz committed:
IFI-2311 Delete modifying-behavior as it's now part of templating section

dsanz committed:
IFI-2311 Rename existing articles to harmonize them with the IFI-2294 requirements

dsanz committed:
IFI-2311 Create the "templating" section skeleton to meet IFI-2294 requirements

dsanz committed:
IFI-2311 Move existing article to the "bundling resources" section as per IFI-2295 requirements

dsanz committed:
IFI-2295 feedback

dsanz committed:
IFI-2311 Rename themelets article as per IFI-2295 requirements

dsanz committed:
IFI-2311 Rename thumbnail article as per IFI-2295 requirements. Add remaining ones (empty)

dsanz committed:
IFI-2311 Add the skeleton for the remaining articles as per IFI-2295 requirements

dsanz committed:
minor changes related to terms used to describe the process

dsanz committed:
use token definitions as token values are not really provided from the theme but via stylebook. Also, consider linking to a broader tooling section

dsanz committed:
try to include blade as a valid option for theme creation, so that reader knows from the beginning that there are 2 flows.

dsanz committed:
I think style books are not a valid mechanism to embed common content, thats what masters are for. We have to find the "official" name for master pages in 7.3

Alec-Shay committed:
Changes per feedback (and fixing filenames + removing comments)

Alec-Shay committed:
Remove obsolete size limitation

Alec-Shay committed:
Flesh out Site Appearance overview article with some barebones content

Alec-Shay commented on docs/dxp/7.x/en/site-building/site-appearance/site-appearance.md:
```diff
@@ -12,46 +12,33 @@ The design and functionality of a Liferay site can be adjusted and customized th
 
-Coming soon!
-<!-- TODO:
+[Master Page Templates](../creating-pages/defining-headers-and-footers/master-page-templates.md) allow you to define elements for all of your pages using Fragments, especially headers and footers. You can use them to conform any number of pages on your Site to the look and feel you design.
```
I wanted to specifically point out this article, since it was largely just a placeholder before -- but if we're going to merge it, then it needs at least some barebones content in it, I think.

I sort of struggled to write this (as I felt I was repeating myself a lot, as far as "different ways to customize your Site"). If you have any thoughts as to how this is presented, or if you'd like to see something different as a starting point for this, please feel free to share them.

dsanz replied:
I can think of cutting the text a little bit:

_"allow you to define common elements for your pages using Fragments, especially headers and footers. "_

Does that make sense?

Alec-Shay replied:
Yeah, that makes sense. I can go with that

dsanz commented on docs/dxp/7.x/en/site-building/site-appearance/themes/theme-development/working-with-templates/embedding-widgets-via-templates.md:
```diff
@@ -1,4 +1,4 @@
-# Embedding Widgets in Your Theme
+# Embedding Widgets via Templates
 
 You can embed a widget in your theme to make it available on all pages using the chosen template file. When you are embedding a widget, you will need to reference the widget's portlet definition <!-- Add link to article explaining portlet definitions in DXP if we will have one? --> to provide some of the component's values.
```
I like this naming!!

dsanz commented on docs/dxp/7.x/en/site-building/site-appearance/themes/theme_development.rst:
```diff
@@ -18,10 +18,10 @@ Styling
 Coming soon!
 
-Templating
+Working with Tempaltes
```
Templates

dsanz commented on docs/dxp/7.x/en/site-building/site-appearance/site-appearance.md:
```diff
@@ -12,46 +12,33 @@ The design and functionality of a Liferay site can be adjusted and customized th
 
```
We may want to use "Master Pages" or "Master Page Templates" for the table headings, rather than "Page Templates", as the latter has a way older meaning in liferay. What matters here is the 'master' term

Alec-Shay replied:
I can change that.

To be clear, though -- does this mean that these things would only apply to Master Page Templates though, and not regular Page Templates?

dsanz replied:
Being it true that page templates offer a way to add elements to a page made from it, I would say that we never introduced them as a customization mechanism like master pages + style books are being introduced in 7.3.

We may want, nevertheless, to mention the concept of page template (and link to the corresponding articles for content and widget page templates) within the Master Page Templates section, to help reader to differentiate bot things, yet keeping the intent of presenting the pack "Masters + stylebooks" in a more appealing and cohesive way.

dsanz commented on docs/dxp/7.x/en/site-building/site-appearance/site-appearance.md:
```diff
@@ -12,46 +12,33 @@ The design and functionality of a Liferay site can be adjusted and customized th
-## Widget/Fragment Appearance
-
-[Maybe show a considerably customized-looking widget/fragment?]
+[Style Books](./style-books/using-a-style-book-to-standardize-site-appearance.md) are sets of visual standards that apply to a Site, such as standard colors and spacing. This allows you to design every page's layout independently while maintaining a consistent experience between each of them.
```
Whatever we write here, it should just be a very brief description taken from the linked page. I'm assuming that's the case here

dsanz commented on docs/dxp/7.x/en/site-building/site-appearance/site-appearance.md:
```diff
@@ -12,46 +12,33 @@ The design and functionality of a Liferay site can be adjusted and customized th
-* Fragments/widgets use themes, and can also be embedded within the themes if necessary
-* Page templates use themes for styling
-* Style books rely on themes for their token definitions
+[Themes](./themes/introduction-to-themes.md) are a highly flexible tool for customizing the default look and feel of your Site. They provide a foundation that your [Style Books](#style-books) and [Page Templates](../creating-pages/adding-pages/creating-a-page-template.md) build on top of. Fragments and themes can also be embedded in the themes themselves to ensure a particular arrangement.
```
I'm not sure about this fragment: "Fragments and themes can also be embedded in the themes themselves to ensure a particular arrangement"

I think that it "Fragment" shall be changed by "Widget". Regarding "themes", I'm not sure what you mean. Are you referring to theme inheritance? Maybe "templates" instead?

Alec-Shay replied:
I think that was a mix-up on my part, and I meant to say "fragments and widgets can also be embedded in themes." (It is true that fragments can be embedded in the themes, too, right?)

dsanz replied:
I am not 100% sure about being able to add fragments to the theme. I've found a taglib to render a fragment, which _theoretically_, shall be usable from a template in a theme very much like the runtime tag we use for widgets. I'd still refrain from mentioning this as I saw no usages of it yet.
But maybe in the future we'll have to revisit this if we see interest in this feature

Alec-Shay committed:
Changes per feedback

Alec-Shay commented:
@dsanz I went ahead and pushed an update with the changes we discussed above. Thanks for all the feedback.

Do you think this reasonably addresses these points -- and, for that matter, is there anything else you would like to discuss with these? Or, do you think we might be ready for this to be our first publicly available draft of the frontend content on learn.liferay.com (and send this forward to JR)?

dsanz commented:
Hi Alec, I think this can be sent to JR for review.
Best
